openbase/bco
# Feature/126 openhab device manager reconnection issues

### :scroll: Description

Changes proposed in this pull request:
* transform openhab connection handling from java to kotlin.
* Fix connection issues with openhab after connection was reestablished.
* Improve exception handling.
* Fix null handling that sometimes leads to null pointer in the connection management.

I would recommend to review the following range in more detail to exclude the kotlin transformation.
https://github.com/openbase/bco/pull/132/files/4b9faa9d6908012dd805837ade0c7c25dd7df8ec..15243da9f5022000b93385fa13060782935460f0

### Test Procedure
1. Start openhab
2. Start `bco-device-manager-openhab`
3. Make sure you receive device updates
4. Make sure you can control devices
5. Stop Openhab
6. Start Openhab
7. Make sure you receive device updates
8. Make sure you can control devices
9. Stop  `bco-device-manager-openhab`
10. Start `bco-device-manager-openhab`
11. Make sure you receive device updates
12. Make sure you can control devices
